components/style/values/computed/mod.rs

```diff
@@ -494,6 +495,24 @@ impl SVGPaint {
 /// <length> | <percentage> | <number>
 pub type LengthOrPercentageOrNumber = Either<Number, LengthOrPercentage>;
 
+#[cfg(feature = "gecko")]
+impl ToAnimatedValue for LengthOrPercentageOrNumber {
```

Comment:
To mitigate the glitches, should we convert LengthOrPercentage to Number in ToAnimatedValue?